### PR TITLE
PHOENIX-5553: Transactional views should not get GlobalIndexChecker c…

### PR DESCRIPTION
**Pull request by gokceni**
…oproc

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java [hunk 1]
```diff
@@ -927,7 +928,16 @@ private void addCoprocessors(byte[] tableName, TableDescriptorBuilder builder, P
...
-            if ((tableType == PTableType.INDEX && !isTransactional) || isViewIndex) {
+            boolean isViewBaseTransactional = false;
+            if (!isTransactional && isViewIndex) {
+                if (tableProps.containsKey(TRANSACTIONAL) &&
+                        Boolean.TRUE.equals(tableProps.get(TRANSACTIONAL))) {
+                    isViewBaseTransactional = true;
+                }
+            }
+
+            if (!isTransactional && !isViewBaseTransactional
```
Comment (kadirozde) [inline phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java]: Do we need to introduce a separate variable, i.e., isViewBaseTransactional? Would not the following work?
If (!isTransactional && (PTableType.INDEX || isViewIndex) {

I think we have a test to check coprocessors on a table for the new index design. Can we update the test for view indexes too?

File: phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java [hunk 2]
```diff
@@ -927,7 +928,16 @@ private void addCoprocessors(byte[] tableName, TableDescriptorBuilder builder, P
             boolean isViewIndex = TRUE_BYTES_AS_STRING
                     .equals(tableProps.get(MetaDataUtil.IS_VIEW_INDEX_TABLE_PROP_NAME));
 
-            if ((tableType == PTableType.INDEX && !isTransactional) || isViewIndex) {
+            boolean isViewBaseTransactional = false;
+            if (!isTransactional && isViewIndex) {
+                if (tableProps.containsKey(TRANSACTIONAL) &&
+                        Boolean.TRUE.equals(tableProps.get(TRANSACTIONAL))) {
```
Comment (BinShi-SecularBird) [inline phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java]: Is "Boolean.TRUE.equals(tableProps.getOrDefault(TRANSACTIONAL, FALSE))" more efficient?

File: phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java [hunk 3]
```diff
@@ -927,7 +928,16 @@ private void addCoprocessors(byte[] tableName, TableDescriptorBuilder builder, P
...
+                    isViewBaseTransactional = true;
```
Comment (BinShi-SecularBird) [inline phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java]: The whole logic is a mess which should be cleaned. I don't understand why we need to check "Boolean.TRUE.equals(tableProps.getOrDefault(TRANSACTIONAL, FALSE))" particularly for viewIndex? Shouldn't "isTransactional = (provider != null) && Boolean.TRUE.equals(tableProps.getOrDefault(TRANSACTIONAL, FALSE))" be applied to both INDEX and viewIndex, and simplify "if (!isTransactional && !isViewBaseTransactional && (tableType == PTableType.INDEX || isViewIndex))" as (!isTransactional && (tableType == PTableType.INDEX || isViewIndex))?

**Comment by gjacoby126** [comment]
JIRA already committed, closing.

**Pull request closed without merging**